core/src/main/java/org/apache/iceberg/TableScanContext.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Collection;
+import java.util.Map;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.Expressions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+
+/**
+ * Context object with optional arguments for a TableScan.
+ */
+final class TableScanContext {
+  private final Long snapshotId;
+  private final Expression rowFilter;
+  private final boolean ignoreResiduals;
+  private final boolean caseSensitive;
+  private final boolean colStats;
+  private final Collection<String> selectedColumns;
+  private final ImmutableMap<String, String> options;
+  private final Long fromSnapshotId;
+  private final Long toSnapshotId;
+
+  TableScanContext() {
+    this(null, Expressions.alwaysTrue(), false, true, false, null, ImmutableMap.of(), null, null);
```

Comment:
I think it would be more clear and maintainable if this didn't use `this(...)`. When new options are added, this list will need to change and it is easy to get off by one when passing in nulls. If instead this set each field default individually, it would be easier to read and avoid update errors.